### DISCUSSION TITLE
[Help Wanted] Understanding editor best-practices; fonts; long-form writing

### DISCUSSION THREAD

**Discussion by X1SDNA** [standardnotes/forum#2709] [General]
Hi

I'm hoping someone can help me understand the best way to use Standard Notes for some common tasks.

In a nutshell, this is a usability and performance question. I'm having troubles with determining the 'cleanest' setup.

Long form writing (eg. long articles, books etc):

This requires a simple setup. However, I have a lot of problems setting font that I want to see when I'm using the editor. When I use the Bold Editor, I can select a font, but it only works sometimes. If I set everything to Times New Roman, then I create an unordered list somewhere in there, it won't display as the font I set. It adds a monospace font. But I can easily just use the Plus Editor instead.

A bigger question that when I click to view the HTML, it wraps every single snippet in inline HTML/CSS styles, like so:

_(Edit - I put the code output here but GitHub just renders it as non-code - have no idea how to show the code - code tags don't show the code - sorry, haven't posted content like this before)_

Is this HTML code only generated on request by clicking the code button, and not actually stored as data? If not, I'm not sure what the purpose of such bloat would be. I'd like to know this so that my brain can be aware of whether I'm creating a whole bunch of mess in my notes just by typing in a rich text editor (or not 💯 )

Although I prefer to keep it simple, I need to use more advanced editors so that I can yellow-highlight text (<mark> in markdown, and background color in rich text editors). This means I have to use the Markdown Pro editor, and rich text editors for all other notes.

In the Markdown Pro editor, I can simply wrap the whole note in a div style and specify the system font to render. So I am wondering why the rich text editors tag absolutely everything with inline styles, instead of doing the same by way of an overall note font setting? Even if this is 'complimentary generator', it still gives insight into how the editors see/apply styling.

So yeah, I'm just not sure what editor to us for long form writing (and also short note taking with a text highlight option), while keeping everything as light, clean, simple and reliable as possible.

To summarize, I believe I need to:

1. Use the Plus Editor for long form content.
2. Use Markdown Pro for shorter notes, because it has the <mark> feature enabled.
3. Find out whether the rich text editors are adding HTML/CSS bloat, or whether that is simply an on-demand generator service.
4. Find out if defaulting all my note editors to the Plus Editor will cause performance or bloat issues...?

If anyone can chime in with their recommendations/experiences, I'll be good to go.

Thanks.

**Comment by JaspalSuri** [comment]
>(Edit - I put the code output here but GitHub just renders it as non-code - have no idea how to show the code - code tags don't show the code - sorry, haven't posted content like this before)

You can use six backticks in two pairs of three to format multiple lines of code as such. 🙂 

```
print("Hello World");
// This is a comment.
```

>Use Markdown Pro for shorter notes, because it has the feature enabled.

Which feature? I wasn't able to locate the text highlighting feature that you were referring to.

>Find out whether the rich text editors are adding HTML/CSS bloat, or whether that is simply an on-demand generator service.

While I wouldn't call it bloat 😄 , the Plus and Bold editors are meant to edit and render HTML, so the underlying/raw/plain text is in HTML.

>Find out if defaulting all my note editors to the Plus Editor will cause performance or bloat issues...?

It won't, but if you were to switch editors, then it could cause issues as you'd have to first copy all of the note's content while using the Plus editor and then paste it into the editor that you're switching to (unless of course it's the Bold Editor).

**Comment by X1SDNA** [comment]
Hi Jaspal - thanks for your response.

> Which feature? I wasn't able to locate the text highlighting feature that you were referring to.

Ah, I see what the GitHub editor did - I didn't have my reference wrapped in code tags. In markdown, the ability to use 'mark' tags to highlight:

```<mark>This text would be highlighted in yellow.</mark>```

 
> While I wouldn't call it bloat 😄 , the Plus and Bold editors are meant to edit and render HTML, so the underlying/raw/plain text is in HTML.

This an example:

```
<p><span style="font-family: 'Times New Roman';"><span style="font-size: 20px;"><strong><span style="font-size: 24px;">This is test for GitHub.</span></strong></span></span></p>

<!--StartFragment--><p><span style="font-family: 'Times New Roman';"><span style="font-size: 20px;"><span style="font-size: 24px;">Lorem ipsum dolor apien eget mi proin. Diam sollicitudin tempor id eu nisl nunc mi ipsum.</span></span></span></p>

<ul><li><span style="font-size: 24px;"><span style="font-family: 'Times New Roman';">Item 1</span></span></li><li><span style="font-size: 24px;"><span style="font-family: 'Times New Roman';">Item 2</span></span></li><li><span style="font-size: 24px;"><span style="font-family: 'Times New Roman';">Item 3</span></span></li><li><span style="font-size: 24px;"><span style="font-family: 'Times New Roman';">Item 4</span></span></li></ul>

<p><span style="font-family: 'Times New Roman';"><span style="font-size: 20px;"><span style="font-size: 24px;">Vitae tortor condimenrutrum.</span></span></span></p>

<p><span style="font-family: 'Times New Roman';"><span style="font-size: 20px;"><span style="font-size: 24px;">Id venenatis a condimentu...
```
Perhaps useful for a snippet or two, but not needed for general word processing. Can you imagine writing _War & Peace_ with this going on?  So I guess in a nutshell I should've just asked what I should use to get a clean and simple word processor...? Unfortunately without that, I can't really replace 90% of what I do with Standard Notes without worrying about a few other things as you mentioned. I assumed that at least one of the editors available would just be a basic word processor without all the code...

I hope I've explained this better than my original post. I'd appreciate your thoughts on this. Many thanks.

**Comment by JaspalSuri** [comment]
> In markdown, the ability to use 'mark' tags to highlight:

Thanks for letting me know about that! I wasn't aware of that particular feature of the Markdown Pro Editor. 

> I hope I've explained this better than my original post. I'd appreciate your thoughts on this. Many thanks.

That makes a lot of sense; I understand that it would be somewhat difficult to work with when exporting the text to other programs or creating a webpage as the inline CSS can get messy and become difficult to manage as the document increases in size and various styles are applied. I think it depends on your goals for the text. If you simply need to keep the notes formatted for your convenience, then you'll find the versatility that HTML provides over Markdown in the Bold and Plus editors when it comes to changing the font and size of the text. If you need to export the text later, then I would suggest using the editors that format Markdown syntax for both long-form writing and taking small notes.

**Comment by X1SDNA** [comment]
Ok, thanks. Just a quick question regarding FileSafe files then:

Can I open them where they are (eg. access within a note on the desktop app), and edit them, with the changes being saved and synced as I work, and just close it when I'm done; and can move to a different device, open it, and keep working in the same manner?

An example would be a docx file that I keep adding to, working on it regularly.

Or does FileSafe require me to download the file to disk, make the changes, delete the previous version from SN, sync, then upload a new version.

I haven't even set up FileSafe yet, but would like to know the scope of it. I totally understand if it is not designed for this purpose - I will still use it for its purpose - but I thought I would ask  because I would just love a simple word processor extension or work-around, and I'm trying to figure it out. Thanks.

**Comment by JaspalSuri** [comment]
> Or does FileSafe require me to download the file to disk, make the changes, delete the previous version from SN, sync, then upload a new version.

This is currently how you'd do it. It's akin to composing an email draft with attachments that is never sent to anyone. 

> I haven't even set up FileSafe yet, but would like to know the scope of it. I totally understand if it is not designed for this purpose - I will still use it for its purpose - but I thought I would ask because I would just love a simple word processor extension or work-around, and I'm trying to figure it out. Thanks.

Of course! Thank you for understanding.

**Comment by X1SDNA** [comment]
Hi - I've tried the Plus Editor, however for some reason it has reached a point where it refuses to save. If I make any changes via either the WYSIWYG editor or the code, and then leave the note - when I come back, the changes are gone. It reverts back to a certain point and I can't seem to fix it. I've cleared cache and re-loaded; that didn't work.

Even if I delete the entire contents of the note, they will re-appear upon return...

Help appreciated. Thanks. 

---

EDIT: I think I've figured it out - please tell me if I am correct:

In my failure to set up CSS 'rules' for 1st line paragraph indentation, I added a `<script>` tag so as to 1. call jQuery - which I now realize wasn't necessary; and 2. to target paragraphs with the node `<br>` using :contains(). Anyway, I had to delete the note itself completely and recreate it, ensuring that I didn't paste the script back in. Seems OK now.

Would this conclusively be the reason it got stuck?

Note: I am still using `<style>My note styling CSS here, etc</style>` at the beginning of the note in Plus Editor code view. Hope that is OK...?

**Comment by X1SDNA** [comment]
Hi - a couple more questions relating to the same topic (setting up the editors for writing):

I need paragraphs first-line indented, except where there is an extra space `ENTER`, in which case there should be no indent. This is turning out to be a huge challenge that I've spent hours on - I definitely need some help!

I have two editor options for this type of work: Plus Editor and Bold Editor.

It should be quite easy to set up each document/note by placing the following at the top of the page in code view:

```
<style>
p {CSS styles go here; text-indent: 2em;}
p:empty + p {text-indent: 0;}
</style>
```

**Plus Editor problems:** Hitting an extra `ENTER` key actually wraps a `br` in `p` tags, so I can't use `p:empty` to select it to apply different indent styling. The editor allows me to target `br` as CSS, but because it is wrapped in a  `p`, I can't figure out a way to make use of this as a selector in order to select the paragraph that immediately follows for non-indentation. I need a CSS expert to let me know how to target the paragraphs immediately following these extra breaks.

**Bold Editor problems:** Although the Bold Editor seems to throw in a random `<br>` sometimes, it generally creates an extra line with `<p></p>`, so the above code works to target it, and I can get the right indent/non-indent styling. Two big problems though: In order for the CSS styling to be applied to the WYSIWYG editor, I need to switch to code view, then switch back, in a sort of 'mini-refresh'. The second issue is a bigger one: the Bold editor refuses to save the `<style>` code - it just strips it out out once I leave the note.

Given that, in addition to stripping styles, the mobile view for the Bold editor leaves almost no room to see anything (menu icons take up a third of the screen, keypad takes up about half - I'm lucky if I see more than a line or two when editing), I think my best bet is to know how to target paragraphs following `<p><br></p>` syntax in Plus Editor using pure CSS.

I'm hoping that there is a way to achieve this seemingly very simple task....?

**Comment by JaspalSuri** [comment]
I see that you've [created a thread for this](https://github.com/summernote/summernote/issues/3863) on the summernote repo; thank you for doing so.

**Comment by X1SDNA** [comment]
Thanks. Yes it was a dead end though. The consensus is that it can't be done via CSS. Given it is a jQuery editor, can you possibly suggest a jQuery line that will solve this? I don't know any jQuery.... Thanks.

**Comment by JaspalSuri** [comment]
I will ask someone from the dev team and get back to you. It might take some time for them to get back to you as they are working on refactoring everything, but I've taken note of it for future reference so that it won't slip through the cracks.
